client/src/modules/reports/generate/stock_entry/stock_entry.config.js

```diff
@@ -22,6 +22,8 @@ function StockEntryConfigController($sce, Notify, SavedReports, AppCache, report
   vm.previewGenerated = false;
   vm.onEntryTypeChange = onEntryTypeChange;
 
+  vm.currency_id = Session.enterprise.currency_id;
```

Comment:
Can you set this to 
```js
vm.reportDetails.currencyId = Session.enterprise.currency_id;
```
instead?  That way, the component is preloaded and can be overridden by a cached currency.